jsonrpc/jsonrpc.go

```diff
@@ -0,0 +1,70 @@
+// Package jsonrpc is a minimal JSON-RPC implementation
+package jsonrpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type JSONRPCRequest struct {
+	ID      interface{}    `json:"id"`
+	Method  string         `json:"method"`
+	Params  [1]interface{} `json:"params"`
+	Version string         `json:"jsonrpc,omitempty"`
+}
+
+func NewJSONRPCRequest(id interface{}, method string, args interface{}) *JSONRPCRequest {
+	return &JSONRPCRequest{
+		ID:      id,
+		Method:  method,
+		Params:  [1]interface{}{args},
+		Version: "2.0",
+	}
+}
+
+// Send sends the request to URL and returns the general JsonRpcResponse, or an error (note: not the JSONRPCError)
+func (req *JSONRPCRequest) Send(url string) (res *JSONRPCResponse, err error) {
```

Comment:
I'd usually expect `Send` to not be a method on `JSONRPCRequest`, but to just accept it as a parameter.
Any particular reason to go this way?

Reply:
What exactly is your proposal - do you mean that `NewJSONRPCRequest` should execute the request?